### DISCUSSION TITLE
Multi Select Request

### DISCUSSION THREAD

**Discussion by crunnels-alt** [twilio-labs/paste#1249] [Q&A] [answered]
Hello, Elastic SIP Trunking have a Console change that would really benefit from a multi select component. Our product designer on loan has done a great job with what we have but Product leadership would like to weigh the option of waiting to develop for the proper tool. We're trying to make an informed decision on if we want to release with the current design we have or wait for the component (and new Product Designer we recently hired to help decide). The design in question is attached. Creating this discussion topic as a data point of teams interested in this. 

![image](https://user-images.githubusercontent.com/80070649/110031570-ef5de900-7ceb-11eb-8401-be38af3b8be5.png)

**Comment by SiTaggart** [comment]
Hi @crunnels-alt 

To get a better sense of what exactly you maybe thinking, are you envisioning something like [this downshift example](https://www.downshift-js.com/use-multiple-selection)?

If not do you have any prior art or examples you're thinking of?

There are quite a few ways to do multi select so this level of detail might help us identify where on our roadmap is currently is.

**Reply by crunnels-alt** [reply]
![image](https://user-images.githubusercontent.com/80070649/110145347-2f75a800-7d8e-11eb-95d6-a75f39d11165.png)
Here is our current experience which is sort of like the downshift example.

We have a requirement that the customer should be able to click on the _element_ (in the case of my most recent screenshot,  FebACL) and edit the contents of this container object directly. I don't believe the downshift example is great for this experience.

The ideal solution shows the ACL objects that are selected, allows quick editing of the objects directly, allows easy search for unselected ACL objects, and is unobtrusive relative to the other features on the screen (Termination URI, Calls Per Second, etc.)

**Comment by TheSisb** [comment, marked as answer]
Hi @crunnels-alt,

We don't currently provide this functionality, but thank you for the data point! 
If you decide to build this for your team, you may be able to achieve almost identical functionality.  One thing to consider is the accessibility of clicking into a selection to edit the value; I'm not sure how easy that would be for keyboard users or screen-readers to do and it may require some testing. I believe that is why all downshift examples lean into only removing that item on click.  

Here are some other examples that may help you get to what you need using Downshift (which we export as `@twilio-paste/combobox-primitive`:

- https://codesandbox.io/s/github/kentcdodds/downshift-examples?file=/src/downshift/other-examples/gmail/index.js
- https://codesandbox.io/s/github/kentcdodds/downshift-examples?file=/src/downshift/ordered-examples/05-multi-create.js


**Note: Make sure to click this button to get the examples to load correctly:**
![image](https://user-images.githubusercontent.com/1592327/111347771-e824cc80-8655-11eb-9557-d535d6e116af.png)

**Comment by crunnels-alt** [comment]
@TheSisb @SiTaggart hello, I noticed that the roadmap for the combobox Paste component I was looking for was pushed from August until November. Can I chat with someone to understand how Paste components get prioritized and if we're able to come up with a solution for the Combobox sooner than November? I was planning a feature release around this August date for quite a while now.

**Reply by SiTaggart** [reply]
hi @crunnels-alt 

There's definitely a few aspects to the prioritization process for us:

- Team capacity is the most obvious place to start, there is only so much we can do at a given time and unfortunately, for the past couple of months we have been running at roughly 25% of the design capacity that we started the year with due to staffing reasons. We're onboarding new designers so there will be a lag to getting back up to full capacity and that's caused us to shift a number of things around.
- We try to do a quarterly roadmap review with internal teams to understand their next quarters needs. That is then cross referenced with an ease and impact survey we have done in the past for each component we haven't built yet. We can then try and balance team needs across the largest number of internal teams, with impact, ease of design and build, and team capacity
- Lastly is team and UX org BPMs alignment

It's also worth mentioning that our [roadmap](https://paste.twilio.design/roadmap/) is, unless explicitly called out, tentative and subject to change as we do quarterly reprioritization through the year. It's quite rare at Twilio for us to be given hard dependencies from internal teams.

With that all said, there is a pathway to unblocking yourselves as suggested by @TheSisb and building a custom version using Paste primitives and Downshift. There is also Design System Office hours which happen weekly, that you and your design partner can attend to get possible alternatives in absence of that component being available by August.
